lib/msprime.c

```diff
@@ -514,6 +514,7 @@ msp_alloc_populations(msp_t *self)
         self->initial_populations[j].growth_rate = 0.0;
         self->initial_populations[j].initial_size = 1.0;
         self->initial_populations[j].start_time = 0.0;
+        self->initial_populations[j].state = MSP_POP_STATE_ACTIVE;
```

Comment:
I take it from this that all of the specified populations are initially active by default? What would happen if we specified some initial samples from a population that we think is an ancestral one (because, eg., it is the parent pop in a split event later on). From what I can see this wouldn't lead to an error even though it probably should -- the initial state of the ancestral pop is set to INACTIVE in the split event, but it doesn't retroactively check that there aren't already samples from the ancestral pop. (I've just skimmed the code so might be missing something here.)

Reply:
Yes, all populations are initially active by default. If we try to take samples at a time when a populations is inactive we get the MSP_ERR_POPULATION_INACTIVE_SAMPLE error. Rather than try to catch these errors at the start of the simulation we just catch them when they happen - that's why there's no logic here trying to think about samples etc.